### PR TITLE
Containerization and related updates

### PR DESCRIPTION
**Pull request by acrewdson**
Apologies for the variegated nature of this PR. A better way to go would have likely been to split this work into a few different PRs. Instead what happened (I _think_, but @OrlandoSoto could say for sure) is that the initial work on getting the app to run in containers surfaced some significant bitrot that required upgrades to dependencies and other miscellaneous tweaks. Not too surprising since this hasn't been touched in a long time. But that in turn led to changes having to be made to the application code itself to maintain compatibility with the newer deps. So we have a real mix of commits here, and most of them are peripheral to the core work: the addition of the three `Dockerfile`s and the `docker-compose.yml`.

### CODE REVIEW

File: client/nginx/default.conf [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    listen       8080;
+    server_name  localhost;
+
+    location / {
+        root   /usr/share/nginx/html;
+        index  index.html index.htm;
+    }
+
+    location /search/ {
+        proxy_pass {{{ES_END_POINT}}};
```
Comment (rosskarchner) [inline client/nginx/default.conf]: I think you need to allow only harmless HTTP methods here...

```sh
curl -XPUT https://search-new-eval.demo.cfpb.gov/search/im-in-ur-elasticsearch
{"acknowledged":true,"shards_acknowledged":true,"index":"im-in-ur-elasticsearch"}
```

Comment (acrewdson) [reply client/nginx/default.conf]: @rosskarchner It's a great point. I'm assuming this setup came from [here](https://github.com/cfpb/github-wiki-search/pull/113/files#diff-72f3e2d92b15895972bfa32294e38e33L51), but what I don't know is whether the old version of the app in production had a better or more-secure config. @OrlandoSoto might be able to track that down so we could find out if the old version of the app was proxying everything. I don't have access to the box.

Comment (acrewdson) [reply client/nginx/default.conf]: @rosskarchner We confirmed that this matches the existing config, unfortunately, so will leave it as-is for now. But we'll note it as a potential future improvement. Thanks for flagging this.

Comment (OrlandoSoto) [reply client/nginx/default.conf]: 👍

**Pull request merged**